### DISCUSSION TITLE
Quarto website not displaying leaflet maps in HTML pages

### DISCUSSION THREAD

**Discussion by jack-davison** [quarto-dev/quarto-cli#1026] [Q&A] [answered]
**EDIT:** A reproducible example is available here: [https://github.com/jack-davison/quarto-leaflet](https://github.com/jack-davison/quarto-leaflet)

Hello,

I've been writing a Quarto website and notice something strange, and wonder if this is expected behaviour or if something is going wrong. I have some webpages that have leaflet maps, and when I knit those pages they correctly display in the "viewer" pane. However, when I open the corresponding HTML page in the `_site` folder, nothing appears.

What may be going on here?

---

**In Viewer:**

![image](https://user-images.githubusercontent.com/45171616/171166324-2b4356ad-2d62-4a8c-b562-2926913acfd3.png)

**HTML**

![image](https://user-images.githubusercontent.com/45171616/171165257-76bed524-0019-4601-9c02-7b34933435bc.png)

It doesn't matter if I knit a page individually or if I use ```quarto::quarto_render()```, nothing actually appears when the HTML files are opened.

**Comment by jjallaire** [comment]
It may be that leaflet requires being "served" from a web server rather than accessed from the local filesystem.

If you could create a small git repo that exhibits this problem we can take a closer look and verify what the problem is.

**Comment by jjallaire** [comment]
I've attempted to reproduce this locally and the map always appears (even when opening directly from the filesystem). Tested on Chrome, Firefox, and Safari on OS X as well as Chrome and Edge on Windows. If you could post a repro that exhibits this problem we'll take a closer look.

**Reply by jack-davison** [reply]
Hi J.J.,

Find the repo here:

[https://github.com/jack-davison/quarto-leaflet](https://github.com/jack-davison/quarto-leaflet)

The issue appears to be if `code-link: true` **AND** the .qmd file is in a folder. See the "problem" page on the website in the repo, which is in the "docs" directory. Both "okay" and "problem" are identical besides one being in a subdirectory. If I turn it to `code-link: false`, however, both work.

**Comment by jack-davison** [comment]
Find a reproducible example here:

[https://github.com/jack-davison/quarto-leaflet](https://github.com/jack-davison/quarto-leaflet)

The issue appears to be if `code-link: true` **AND** the .qmd file is in a folder. See the "problem" page on the website in the repo, which is in the "docs" directory. Both "okay" and "problem" are identical besides one being in a subdirectory. If I turn it to `code-link: false`, however, both work.

![image](https://user-images.githubusercontent.com/45171616/171180397-dbbfcfbe-23b7-40bc-ab9f-0b793bb3289b.png)

**Comment by jjallaire** [comment, marked as answer]
Okay, thanks so much for your patience and for these repro cases. This was actually two bugs for the price of one! (https://github.com/quarto-dev/quarto-cli/commit/9bb09f5369e9945b11e3c4b55d0c39039ae344a5 and https://github.com/quarto-dev/quarto-cli/commit/8bf851e4e947bfe96b2a677a1873226f97413043)

Fixes are available in 0.9.487: https://github.com/quarto-dev/quarto-cli/releases/latest

**Reply by jack-davison** [reply]
Thanks very much - can confirm the update works on my system!
